src/core/border_router/routing_manager.cpp

```diff
@@ -817,6 +817,45 @@ bool RoutingManager::NetworkDataContainsUlaRoute(void) const
     return contains;
 }
 
+#if OPENTHREAD_CONFIG_BORDER_ROUTING_REACHABLITY_CHECK_ICMP6_ERROR_ENABLE
+
+void RoutingManager::CheckReachbilityToSendIcmpError(const Message &aMessage, const Ip6::Header &aIp6Header)
+{
+    Ip6::MessageInfo messageInfo;
+
+    VerifyOrExit(IsRunning() && IsInitalPolicyEvaluationDone());
+
+    VerifyOrExit(!aIp6Header.GetDestination().IsMulticast());
+
+    // Validate that source matches the ULA OMR prefix (which is not
+    // infrastructure-derived).
+
+    VerifyOrExit(!mOmrPrefixManager.GetFavoredPrefix().IsEmpty());
+    VerifyOrExit(!mOmrPrefixManager.GetFavoredPrefix().IsInfrastructureDerived());
+    VerifyOrExit(aIp6Header.GetSource().MatchesPrefix(mOmrPrefixManager.GetFavoredPrefix().GetPrefix()));
+
+    VerifyOrExit(mRoutePublisher.IsPublishingDefaultRoute());
+
+    VerifyOrExit(!mRxRaTracker.IsAddressOnLink(aIp6Header.GetDestination()));
+    VerifyOrExit(!mRxRaTracker.IsAddressReachableThroughExplicitRoute(aIp6Header.GetDestination()));
+    VerifyOrExit(!Get<NetworkData::Leader>().IsNat64(aIp6Header.GetDestination()));
+
+    LogInfo("Send ICMP unreachable for fwd msg with local ULA src and outside-AIL dst");
+    LogInfo("   src: %s", aIp6Header.GetSource().ToString().AsCString());
+    LogInfo("   dst: %s", aIp6Header.GetDestination().ToString().AsCString());
+
+    messageInfo.Clear();
+    messageInfo.SetPeerAddr(aIp6Header.GetSource());
+
+    IgnoreError(Get<Ip6::Icmp>().SendError(Ip6::Icmp::Header::kTypeDstUnreach,
+                                           Ip6::Icmp::Header::kCodeDstUnreachNoRoute, messageInfo, aMessage));
```

Comment:
According to current text from SPEC-1218 the error code should be 1 - Communication with destination administratively prohibited. 

`If the Border Router needs to forward a received IPv6 packet to a destination that is not on the Adjacent Infrastructure Link, and the IPv6 source address of the packet matches a current ULA OMR prefix with P_preference = ‘11’ (Low), then it MUST send an ICMPv6 Destination Unreachable message with ICMPv6 Code set to '1' (Communication with destination administratively prohibited) back to the source of the packet.`

Reply:
Thanks. Updated in new push.